hail/python/setup.py

```diff
@@ -19,12 +19,24 @@
 
         pkg = stripped
 
-        if pkg.startswith('pyspark') and os.path.exists('../env/SPARK_VERSION'):
-            with open('../env/SPARK_VERSION', 'r') as file:
-                spark_version = file.read()
-            dependencies.append(f'pyspark=={spark_version}')
-        else:
-            dependencies.append(pkg)
+def add_dependencies(fname):
+    with open(fname, 'r') as f:
+        for line in f:
+            stripped = line.strip()
+            if stripped.startswith('#') or len(stripped) == 0:
+                continue
+            if stripped.startswith('-r'):
+                additional_requirements = stripped[len('-r'):].strip()
+                add_dependencies(additional_requirements)
+                continue
+            pkg = stripped
+            if pkg.startswith('pyspark') and os.path.exists('../env/SPARK_VERSION'):
+                with open('../env/SPARK_VERSION', 'r') as file:
+                    spark_version = file.read()
+                dependencies.append(f'pyspark=={spark_version}')
```

Comment:
why shouldn't we error if SPARK_VERSION is missing?

Reply:
Great question, though this isn't a new change just moved code. Can I make a separate tiny pr to nix that check?